### DISCUSSION TITLE
How to change an old version of the source file to a new version of the source file？

### DISCUSSION THREAD

**Discussion by KangChenRui** [idaholab/moose#24082] [Q&A General] [answered]
Hello everyone!

 The old version of the source file have "template <>" in MOOSE.
 How can I convert an old version of the source file into a new version of the source file? Because I made modifications to the source file in the old version, now I want to use the modified source file for calculation.
Thanks for your help！
[new ACInterface.txt](https://github.com/idaholab/moose/files/11241001/new.ACInterface.txt)
[old ACInterface.txt](https://github.com/idaholab/moose/files/11241002/old.ACInterface.txt)

**Comment by GiudGiud** [comment, marked as answer]
Hello

You're likely referring to the "validParams" template deprecation. 
There's more information on updating there
https://mooseframework.inl.gov/newsletter/2020/2020_04.html#c1ba1ef6-13a6-4210-bedb-0b691fec2ce7

From the files it seems you've already done it. What is the issue now?
You can modify the source files as you need

Guillaume

**Reply by KangChenRui** [reply]
Thanks for your reply!
